api-metastore/src/main/java/org/zalando/nakadi/controller/SchemaController.java

```diff
@@ -53,6 +58,49 @@ public SchemaController(final SchemaService schemaService, final EventTypeServic
         return status(HttpStatus.OK).build();
     }
 
+    @RequestMapping(value = "/event-types/{name}/schemas/{version}/compatibility-check", method = RequestMethod.POST)
+    public ResponseEntity<?> checkCompatibility(@PathVariable("name") final String name,
+                                                @PathVariable("version") final String version,
+                                                @Valid @RequestBody final EventTypeSchemaBase schema,
+                                                final Errors errors) {
+        if (errors.hasErrors()) {
+            throw new ValidationException(errors);
+        }
+
+        final EventType eventType = eventTypeService.get(name);
+
+        if(!schema.getType().equals(eventType.getSchema().getType())){
+            final var detail = String.format("schema type cannot be " +
+                            "different: expected `%s`, but was `%s`"
+                    , eventType.getSchema().getType(), schema.getType());
+
+            return status(HttpStatus.CONFLICT).
+                    body(Problem.valueOf(Status.CONFLICT, detail));
+        }
+
+        final var toCompareEventTypeSchema = version.equals("latest")?
+                eventType.getSchema(): schemaService.getSchemaVersion(name, version);
+        eventType.setSchema(toCompareEventTypeSchema);
+
+        final var newEventTypeBase = new EventTypeBase(eventType);
+        newEventTypeBase.setSchema(schema);
+
+        return compatibilityResponse(eventType, newEventTypeBase);
+    }
+
+    private ResponseEntity<?> compatibilityResponse(final EventType eventType, final EventTypeBase newEventTypeBase) {
+        final var compatibilityMode = newEventTypeBase.getCompatibilityMode();
+        final var version = eventType.getSchema().getVersion().toString();
+        try {
+            schemaService.getValidEvolvedEventType(eventType, newEventTypeBase);
+        } catch (SchemaEvolutionException e) {
+            return status(HttpStatus.OK).
```

Comment:
maybe it is a matter of taste, but I prefer to use 2 local vars and assign values:
`compatible = true`
`error = null`

and only sigle return at the end

Reply:
it will also clarify magic contants in the CompatibilityResponse ctr

Reply:
> maybe it is a matter of taste, but I prefer to use 2 local vars and assign values: `compatible = true` `error = null`
> 
> and only sigle return at the end

Since our code style requires all variables to be final, its not possible to reassign variables to new values and only have a single return at the end. Something like below is not possible.

``` java
    private ResponseEntity<?> compatibilityResponse(final EventType eventType, final EventTypeBase newEventTypeBase) {
        final var compatibilityMode = newEventTypeBase.getCompatibilityMode();
        final var version = eventType.getSchema().getVersion().toString();
        var compatible = true;
        String errorMsg = null;
        try {
            schemaService.getValidEvolvedEventType(eventType, newEventTypeBase);
        } catch (SchemaEvolutionException e) {
            compatible = false;
            errorMsg = e.getMessage();
        }
        return status(HttpStatus.OK).
                body(new CompatibilityResponse(compatible, compatibilityMode, errorMsg, version));
    }
```